src/c/vensim.c

```diff
@@ -124,24 +124,27 @@ double _ZIDZ(double a, double b) {
 //
 // Lookups
 //
-Lookup* __new_lookup(size_t size, ...) {
+Lookup* __new_lookup(size_t size, bool copy, double *data) {
 	// Make a new Lookup with "size" number of pairs given in x, y order in a flattened list.
 	Lookup* lookup = malloc(sizeof(Lookup));
-	lookup->data = malloc(sizeof(double) * 2 * size);
 	lookup->n = size;
 	lookup->inverted_data = NULL;
-	// Copy arguments into the lookup data.
-	va_list args;
-	va_start(args, size);
-	for (size_t i = 0; i < 2 * size; i++) {
-		*(lookup->data + i) = va_arg(args, double);
+	lookup->data_is_owned = copy;
+	if (copy) {
+		// Copy array into the lookup data.
+		lookup->data = malloc(sizeof(double) * 2 * size);
+		for (size_t i = 0; i < 2 * size; i++) {
```

Comment:
Why do you use size_t in this for loop but int on Line 262?    Also, I'm not 100%, but I believe at one point WASM handle some number types (double vs float, int vs ???) better than others.  This might be from WebAssembly days and not apply to WASM now.  Emscripten has evolved.

Reply:
In most cases where you're indexing into an array, it's a good idea to use `size_t`, since it's an unsigned type and it matches the address size of the CPU which is important for addressing large data sets.  We use `size_t` nearly everywhere, but the loop at 262 is an anomaly because it's filling an array with index values of type `int`.  I changed that line because the compiler flagged it as a type mismatch warning; could've also fixed it by casting or changing the type of the `ind` field in `dbl_ind` to `size_t`, but figured changing the loop variable to `int` best reflects its usage.